ms/MSOper/MSSummary.cc

```diff
@@ -1048,41 +1023,52 @@ void MSSummary::listField (LogIO& os, Record& outrec,  Bool verbose, Bool fillRe
         }
         os << endl;
         // loop through fields
-        for (uInt i=0; i<fieldId.nelements(); i++) {
-            uInt fld=fieldId(i);
-            if (fld<msFC.phaseDir().nrow()) {
-                MDirection mRaDec=msFC.phaseDirMeas(fld);
+        vector<MDirection> phaseDirs = _msmd->getPhaseDirs();
+        vector<String> fieldNames = _msmd->getFieldNames();
+        vector<String> codes = _msmd->getFieldCodes();
+        std::set<Int>::const_iterator fiter = uniqueFields.begin();
+        std::set<Int>::const_iterator fend = uniqueFields.end();
+        vector<Int> sourceIDs = _msmd->getFieldTableSourceIDs();
+        for (; fiter!=fend; ++fiter) {
```

Comment:
actually nevermind, fiter is the uniquefield int set not the names ... so its sorted the same way